doc/workplane.rst

```diff
@@ -0,0 +1,557 @@
+
+You can select Vertices, Edges, Faces, Solids, and Wires using selectors.
+
+Think of selectors as the equivalent of your hand and mouse, if you were to build an object using a conventional CAD system.
+
+See :ref:`selectors` to learn more.
+
+
+Construction Geometry
+---------------------------
+Construction geometry are features that are not part of the object, but are only defined to aid in building the object.
+A common example might be to define a rectangle, and then use the corners to define the location of a set of holes.
+
+Most CadQuery construction methods provide a ``forConstruction`` keyword, which creates a feature that will only be used
+to locate other features.
+
+
+The Stack
+---------------------------
+
+As you work in CadQuery, each operation returns a new Workplane object with the result of that
+operations. Each Workplane object has a list of objects, and a reference to its parent.
+
+You can always go backwards to older operations by removing the current object from the stack.  For example::
+
+    Workplane(someObject).faces(">Z").first().vertices()
+
+returns a CadQuery object that contains all of the vertices on the highest face of someObject. But you can always move
+backwards in the stack to get the face as well::
+
+    Workplane(someObject).faces(">Z").first().vertices().end()
+
+You can browse stack access methods here: :ref:`stackMethods`.
+
+
+.. _chaining:
+
+Chaining
+---------------------------
+
+All Workplane methods return another Workplane object, so that you can chain the methods together
+fluently. Use the core Workplane methods to get at the objects that were created.
+
+Each time a new Workplane object is produced during these chained calls, it has a
+:attr:`~cadquery.Workplane.parent` attribute that points to the Workplane object that created it.
+Several CadQuery methods search this parent chain, for example when searching for the context solid.
+You can also give a Workplane object a tag, and further down your chain of calls you can refer back
+to this particular object using its tag.
+
+
+The Context Solid
+---------------------------
+
+Most of the time, you are building a single object, and adding features to that single object.  CadQuery watches
+your operations, and defines the first solid object created as the 'context solid'.  After that, any features
+you create are automatically combined (unless you specify otherwise) with that solid.  This happens even if the
+solid was created  a long way up in the stack.  For example::
+
+    Workplane("XY").box(1, 2, 3).faces(">Z").circle(0.25).extrude(1)
+
+Will create a 1x2x3 box, with a cylindrical boss extending from the top face.  It was not necessary to manually
+combine the cylinder created by extruding the circle with the box, because the default behavior for extrude is
+to combine the result with the context solid. The hole() method works similarly -- CadQuery presumes that you want
```

Comment:
Does the `hole()` reference here need to be tied back to the API documentation, or does it matter?